geodiff/src/geodiff.cpp

```diff
@@ -106,6 +106,54 @@ int GEODIFF_createChangesetEx( const char *driverName, const char *driverExtraIn
 }
 
 
+int GEODIFF_createChangesetDr( const char *driverSrcName, const char *driverSrcExtraInfo, const char *src,
+                               const char *driverDstName, const char *driverDstExtraInfo, const char *dst,
+                               const char *changeset )
+{
+  if ( !driverSrcName || !driverDstName || !src || !dst || !changeset )
+  {
+    Logger::instance().error( "NULL arguments to GEODIFF_createChangesetAcrossDrivers" );
+    return GEODIFF_ERROR;
+  }
+
+  if ( strcmp( driverSrcName, driverDstName ) == 0 )
+  {
+    return GEODIFF_createChangesetEx( driverSrcName, driverSrcExtraInfo, src, dst, changeset );
+  }
+
+  // copy both sources to geopackage and create changeset
+  TmpFile tmpSrcGpkg( "" );
```

Comment:
can we have constuctor with no arguments for TmpFile, so we do not need this ugly call?